src/azure-cli/azure/cli/command_modules/mysql/custom.py

```diff
@@ -58,6 +59,128 @@ def server_list_custom_func(client, resource_group_name=None):
     return client.list()
 
 
+def flexible_server_threat_model_update(cmd, client, resource_group_name,
+                                        server_name,
+                                        defender_state=None,
+                                        advanced_threat_protection_name="default",
+                                        subscription_id=None
+                                        ):
+    if resource_group_name is None:
+        raise ValueError("Invalid resource group name provided.")
+
+    if server_name is None:
+        raise ValueError("Invalid server name provided.")
+
+    if advanced_threat_protection_name is None:
+        raise ValueError("Invalid defender protection name provided.")
+
+    if defender_state is None:
+        raise ValueError("Invalid defender state provided.")
+
+    # parameters = {
+    #     'state': defender_state
+    # }
+
+    response = {
+        "id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/threatprotection-4799/providers/Microsoft.DBforMySQL/flexibleServers/threatprotection-6440/advancedThreatProtectionSettings/Default",
+        "name": "Default",
+        "type": "Microsoft.DBforMySQL/flexibleServers/advancedThreatProtectionSettings",
+        "systemData": {
+            "createdBy": "string",
+            "createdByType": "User",
+            "createdAt": "2022-04-03T04:41:33.937Z",
+            "lastModifiedBy": "string",
+            "lastModifiedByType": "User",
+            "lastModifiedAt": "2022-04-03T04:41:33.937Z"
+        },
+        "properties": {
+            "state": "{fname}",
+            "creationTime": "2022-04-03T04:41:33.937Z",
+            "provisioningState": "Succeeded"
+        }
+    }
+
+    if defender_state == "Enabled":
```

Comment:
You should directly use the defender_state variable to assign the value to the property, there is no need to use a if-else